tools/locate_bins.sh

```diff
@@ -0,0 +1,57 @@
+#!/bin/sh -e
+# Copyright (c) 2017-2022, Mudita Sp. z.o.o. All rights reserved.
+# For licensing, see https://github.com/mudita/MuditaOS/LICENSE.md
+
+have_command () {
+  eval "$1 --version >> /dev/null 2>&1"
+  if [ $? -ne 0 ]
+  then
+    echo "false"
+  fi
+  echo "true"
+}
+
+eval "sudo -n echo >> /dev/null 2>&1"
+if [ $? -eq 0 ]; then
+  export have_sudo="true"
+else
+  export have_sudo="false"
+fi
+
+get_command () {
+  path_val=""
+  if $(have_command $1) ]; then
+    path_val=$(command -v $1)
+  elif $have_sudo; then
+    path_val=$(sudo -l $1 2>&1)
+    if [ "$path_val" ]; then
+      if [ -e "$path_val" ]; then
+        cmd_val="sudo $path_val"
+      else
+        return 1
+      fi
+    else
+      return 2
+    fi
+  else
+    return 1
+  fi
+  eval "export -p call_${1}=\"$path_val\""
+  return 0
+}
+
+require_commands () {
```

Comment:
what programs did you not have installed? these should be all installed via `bootstrap.sh` therefore checking for existence, in theory, shouldn't be required.
This is actually pretty important as bootstrap was used to create docker images too so that we have docker images with all tools required.

Reply:
I have all the programs installed, the problem is that they have to be invoked with root level permissions.  This is why I check that they can be invoked or invoked using sudo.  Checking that they are installed is merely a reflection of what the tools and tests scripts already do.

Reply:
Did you check how it's done in bootstrap? do we really require to create bash wrappers just for wrappers?
These can be run with root, but you can add them to your sudo config to run them without sudo, add group, etc. to not use root permissions etc.

Reply:
I have checked bootstrap script and it simply uses `sudo` without 
* checking if `sudo` is installed
* checking if `sudo` is required for a particular command
* checking if you the right to run a program using `sudo`
* checking if `sudo` will run non-interactively (it will prompt you literally five times for the same password in the boostrap script)

The when making the command wrappers, the script checks if `sudo` is installed but **only** uses it when it is needed (current user lacks rights), allowed (you are permitted via sudoers), and non-interactive (so you don't get prompted 20 times per build).